go-controller/pkg/ovn/controller/apbroute/network_client.go

```diff
@@ -507,15 +512,20 @@ func (nb *northBoundClient) ensureRouteInfoLocked(podName ktypes.NamespacedName)
 	// 5. If routeInfo was deleted between releasing the cache lock and grabbing
 	// the routeInfo lock, return an error so the caller doesn't use it and
 	// retries the operation later
-	if routeInfo.Deleted {
-		routeInfo.Unlock()
-		return nil, fmt.Errorf("routeInfo for pod %s, was altered during ensure route info", podName)
+	if nb.externalGatewayRouteInfo.GetRouteInfoDeletedStatus(podName) {
+		if !isDeleted {
+			// info was modified while waiting for unlock, return error and retry later
+			routeInfo.Unlock()
+			return nil, fmt.Errorf("routeInfo for pod %s, was altered during ensure route info", podName)
+		}
+		// it was already deleted before the lock, so change the status as not deleted
```

Comment:
So maybe this has changed in this PR, but I was curious what handles deleting entries out of the cache, and I saw this:

```
func (oc *DefaultNetworkController) deleteRouteInfoLocked(name ktypes.NamespacedName) *apbroutecontroller.ExternalRouteInfo {
	// Attempt to find the routeInfo in the cache, release the cache lock while
	// we try to lock the routeInfo to avoid any deadlock
	oc.exGWCacheMutex.RLock()
	routeInfo := oc.externalGWCache[name]
	oc.exGWCacheMutex.RUnlock()

	if routeInfo == nil {
		return nil
	}
	routeInfo.Lock()

	if routeInfo.Deleted {
		routeInfo.Unlock()
		return nil
	}

	routeInfo.Deleted = true

	go func() {
		oc.exGWCacheMutex.Lock()
		defer oc.exGWCacheMutex.Unlock()
		if newRouteInfo := oc.externalGWCache[name]; routeInfo == newRouteInfo {
			delete(oc.externalGWCache, name)
		}
	}()

	return routeInfo
}
```

this comparison doesnt look right "routeInfo == newRouteInfo". That looks like its comparing 2 pointers to structs. Dont you want deep equal?